tests/bluetooth/bsim_bt/bsim_test_audio/src/mcc_test.c

```diff
@@ -570,11 +569,12 @@ static void connected(struct bt_conn *conn, uint8_t err)
 	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
 	if (err) {
+		bt_conn_unref(conn);
+		default_conn = NULL;
```

Comment:
It might be more appropriate to do `bt_conn_unref(default_conn)` since if I understand correctly that's the reference that you want to unref (and the assumption is that `conn == default_conn`).